code/controllers/subsystem/movement/movement.dm

```diff
@@ -86,22 +88,33 @@ SUBSYSTEM_DEF(movement)
 	buckets -= "[bucket_time]"
 
 /datum/controller/subsystem/movement/proc/queue_loop(datum/move_loop/loop)
-	var/target_time = loop.timer
-	var/string_time = "[target_time]"
+	if(loop.queued_time != -1)
+		stack_trace("just tried to add a loop at the time ([loop.timer]) that already has a queued time ([loop.queued_time]), what'd you do?")
+		return
```

Comment:
`ASSERT` would work better here

Reply:
true. I'm a bit concerned because this doesn't actually catch what it should, needs further debugging